components/MyMessage.vue

```diff
@@ -790,6 +801,9 @@ export default {
       const router = useRouter()
       router.push(this.message.type === 'Offer' ? '/give' : '/find')
     },
+    repostWhenUnavailable() {
+      this.triedToRepost = true
```

Comment:
Suppose we go through here, and then wait until a repost is allowed.  Will they see the right things happen?

Reply:
I think with the code as you've written it, they wouldn't be able to repost and would need to refresh the page to become able to do so.  But they don't know that they need to do that.  This isn't a very likely case, but I don't want them to get stuck.

I think if you fetch the message again from the server (using `force` flag) inside `repostWhenUnavailable` then you will have an up to date value of `canrepost`. If that is now `true` then you can call `repost`. 

That means if they stay on the page and click the button at what looks to them like a sensible time, the repost will happen.

Reply:
@edwh updated. Hopefully, I understood correctly what you meant.